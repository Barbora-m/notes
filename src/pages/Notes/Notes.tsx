import { Button, Col, Container, Row, Spinner } from "reactstrap";
import { useTranslation } from "react-i18next";
import { Link } from "react-router-dom";
import styled from "styled-components";

import { useFetch } from "../../hooks";
import { Note as NoteType } from "../../types";

export const Notes = () => {
  const { t, i18n } = useTranslation();

  const { response: notes, isLoading } = useFetch<Array<NoteType>>({
    url: "/",
    method: "GET",
  });

  return (
    <Container>
      <Languages>
        <LanguageButton size="sm" onClick={() => i18n.changeLanguage("en")}>
          English
        </LanguageButton>
        <LanguageButton size="sm" onClick={() => i18n.changeLanguage("cz")}>
          ÄŒesky
        </LanguageButton>
      </Languages>
      <Row>
        <Header>
          <h4>{t("notes")}</h4>
          <Link to="/notes/add">
            <Button size="sm" color="success">
              {t("addNote")}
            </Button>
          </Link>
        </Header>
      </Row>
      {isLoading && <Loader color="success" />}
      <NotesWrapper>
        {notes?.map(({ id, content }) => {
          return (
            <NoteLink key={id} to={`/notes/${id}`}>
              {content}
            </NoteLink>
          );
        })}
      </NotesWrapper>
    </Container>
  );
};

const Languages = styled.div`
  display: flex;
  justify-content: flex-end;
`;

const LanguageButton = styled(Button)`
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  margin-right: 5px;

  &:focus {
    box-shadow: none;
  }

  &:last-child {
    margin-right: 0;
  }
`;

const Header = styled(Col)`
  display: flex;
  justify-content: space-between;
  margin-top: 20px;
  padding-left: 25px;
`;

const Loader = styled(Spinner)`
  display: block;
  margin: 0 auto;
`;

const NotesWrapper = styled.div`
  padding-bottom: 20px;
`;

const NoteLink = styled(Link)`
  border-bottom: 1px solid darkgray;
  color: black;
  display: block;
  height: auto;
  padding: 10px;

  &:last-child {
    border: none;
  }

  &:hover {
    background: rgba(0, 0, 0, 0.1);
    color: black;
    text-decoration: none;
    transition: background-color 0.25s;
  }
`;
