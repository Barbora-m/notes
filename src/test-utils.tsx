import { ReactElement } from "react";
import { I18nextProvider } from "react-i18next";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import { i18n } from "./i18n";

export const renderWithProviders = (component: ReactElement) =>
  renderer.create(
    <I18nextProvider i18n={i18n}>
      <MemoryRouter>{component}</MemoryRouter>
    </I18nextProvider>
  );
