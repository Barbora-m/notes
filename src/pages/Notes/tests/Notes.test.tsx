import { renderWithProviders } from "../../../test-utils";
import { Notes } from "../Notes";

test("Notes", () => {
  expect(renderWithProviders(<Notes />)).toMatchSnapshot();
});
