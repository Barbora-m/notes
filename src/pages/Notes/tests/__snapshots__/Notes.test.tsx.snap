// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes 1`] = `
<div
  className="container"
>
  <div
    className="sc-bdfBwQ gZMobo"
  >
    <button
      aria-label={null}
      className="sc-gsTCUz dOHhxu btn btn-secondary btn-sm"
      onClick={[Function]}
      type="button"
    >
      English
    </button>
    <button
      aria-label={null}
      className="sc-gsTCUz dOHhxu btn btn-secondary btn-sm"
      onClick={[Function]}
      type="button"
    >
      ÄŒesky
    </button>
  </div>
  <div
    className="row"
  >
    <div
      className="sc-dlfnbm lawOwT col"
    >
      <h4>
        Notes
      </h4>
      <a
        href="/notes/add"
        onClick={[Function]}
      >
        <button
          aria-label={null}
          className="btn btn-success btn-sm"
          onClick={[Function]}
        >
          Add note
        </button>
      </a>
    </div>
  </div>
  <div
    className="sc-eCssSg enUuWu"
  />
</div>
`;
